subsys/net/lib/nrf_cloud/src/nrf_cloud_fota_poll.c

```diff
@@ -563,14 +583,19 @@ int nrf_cloud_fota_poll_process(struct nrf_cloud_fota_poll_ctx *ctx)
 		return -EAGAIN;
 	}
 
+	if (ctx_ptr->status_fn) {
+		ctx_ptr->status_fn(NRF_CLOUD_FOTA_DOWNLOADING, NULL);
+		k_work_cancel_delayable(&ctx_ptr->timeout_work);
```

Comment:
is the cancel needed here?  if so, should it be moved to the top of the function since the job maintenance could potentially take a few seconds.